docs/learn/workflow.md

```diff
@@ -0,0 +1,222 @@
+
+Like every scientific software, the `nyaml` tool also follows a specific workflow.
+
+```mermaid
+graph TD;
+  subgraph Start
+    id1["Input File (YAML or XML)"]
+  end
+  subgraph Correct File Converter
+    id2["nyaml2nxdl Converter"]
+    id3["nxdl2nyaml Converter"]
+  end
+  subgraph nyaml2nxdl
+    id4["Comment Collector"]
+    id5["Python Dictionary Object"]
+  end
+  subgraph nxdl2nyaml
+    id6["XML Object"]
+  end
+  subgraph Final result
+  id7["Write XML File"]
+  id8["Write YAML File"]
+  end
+
+  id1--> |YAML File|id2
+  id1--> |XML File|id3
+  id2-->id4
+  id4-->id5
+  id3-->id6
+  id5-->id7
+  id6-->id8
+```
+
+## Conversion from YAML to XML and vice versa
+Presented below is a concise and trimmed example of the `NXmpes` application definition (for discussion purpose) in `YAML` format, alongside its corresponding encoding into `XML` format, as illustrated below. Subsequently, the fundamental rules governing this conversion process are elucidated. For a comprehensive understanding of the basic structure of NXDL, readers are encouraged to explore the [NeXus Manual](https://manual.nexusformat.org/user_manual.html).
+
+!!! note
+    For in detailed explanation with examples please follow the [Tutorials for writing NeXus definition in YAML](../tutorials/tutorials.md).
+!!! note
+    The `NXmpes` application is an old and trimmed version which does not serve any scientific purpose, but only is used to illustrate the results of the conversion process.
+
+<!-- To do, add warning saying that the application definitions is not mandatory should the update NXmpes application. -->
+
+**NXmpes application definition in YAML and XML format**
+
+=== "YAML"
+    ```yaml
+    category: application
+    type: group
+    doc: |
+      This is the most general application definition for multidimensional photoelectron spectroscopy.
+
+      .. _ISO 18115-1:2023: https://www.iso.org/standard/74811.html
+      .. _IUPAC Recommendations 2020: https://doi.org/10.1515/pac-2019-0404
+    symbols:
+      doc: |
+        The symbols used in the schema to specify e.g. dimensions of arrays
+      n_transmission_function: |
+        Number of data points in the transmission function.
+    NXmpes(NXobject):
+      (NXentry):
+        definition:
+          \@version:
+          enumeration: [NXmpes]
+        title:
+        start_time(NX_DATE_TIME):
+          doc: |
+            Datetime of the start of the measurement.
+        end_time(NX_DATE_TIME):
+          exists: recommended
+          doc: |
+            Datetime of the end of the measurement.
+        (NXinstrument):
+          doc:
+          - |
+            Description of the MPES spectrometer and its individual parts.
+          - |
+            xref:
+              spec: ISO 18115-1:2023
+              term: 12.58
+              url: https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:term:12.58
+          source_TYPE(NXsource):
```

Comment:
nameType should be added even though this being an old example

Reply:
Added!